🐛 Modify SSE connection handling for Puma

terraboops opened the pull request:
- Keep track of events on per-client basis
- Clients get a UUID
- If a client's event queue grows too big it gets dropped
- If a Puma thread tries to send the next batch of events to a client and it's event queue is dropped, the queue will be created
- Updated tests to match above changes

tylermauthe commented on lib/dashing/app.rb:
```diff
@@ -72,10 +74,15 @@ def authenticated?(token)
+      while event = settings.connection_events[connection_id].shift do
+        out << event unless out.closed?
+      end
+    end
```
This loop is killed by Puma when connection is terminated -- in fact the method doesn't even complete. The `GET` request to `/events` is still logged by Sinatra, so on some level the application is aware of the connection being closed -- it would be great if the `connection_id` was removed from the set of `connnection_events` at this time but I couldn't figure out how to do this.

Commit:
:bug: Modify SSE connection handling for Puma

- Keep track of events on per-client basis
- Clients get a UUID
- If a client's event queue grows too big it gets dropped
- If a Puma thread tries to send the next batch of events to a client and it's event queue is dropped, the queue will be created
- Updated tests to match above changes
